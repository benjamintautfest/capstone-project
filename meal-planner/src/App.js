import { useState } from 'react'
import styled from 'styled-components/macro'
import woodBackground from './assets/wood_background.jpg'
import logo from './assets/logo.svg'
import Button from './components/Button'
import MealSelectMenu from './components/MealSelectMenu'
import { v4 as uuidv4 } from 'uuid'

export default function App() {
    const [weekdays, setWeekdays] = useState([
        { weekday: 'Montag', id: uuidv4(), meal: '' },
        { weekday: 'Dienstag', id: uuidv4(), meal: '' },
        { weekday: 'Mittwoch', id: uuidv4(), meal: '' },
        { weekday: 'Donnerstag', id: uuidv4(), meal: '' },
        { weekday: 'Freitag', id: uuidv4(), meal: '' },
        { weekday: 'Samstag', id: uuidv4(), meal: '' },
        { weekday: 'Sonntag', id: uuidv4(), meal: '' },
    ])

    const [selectedWeekday, setSelectedWeekday] = useState(null)

    function handleClick(event) {
        setSelectedWeekday(event.currentTarget.id)
    }

    function selectMeal(meal, id) {
        setSelectedWeekday(null)
        const updatedWeekdays = weekdays.map((weekday) =>
            weekday.id === id ? { ...weekday, meal } : weekday
        )
        setWeekdays(updatedWeekdays)
    }

    return (
        <AppStyled>
            <header>
                <img src={logo} alt="" />
            </header>
            <section>
                {weekdays.map(({ weekday, id, meal }) => (
                    <div>
                        <Button
                            day={weekday}
                            id={weekday}
                            meal={meal}
                            onClick={handleClick}
                            key={id}
                        />
                        {selectedWeekday === weekday && (
                            <MealSelectMenu
                                handleMealClick={selectMeal}
                                weekdayId={id}
                            />
                        )}
                    </div>
                ))}
            </section>
            <footer></footer>
        </AppStyled>
    )
}

const AppStyled = styled.div`
    display: grid;
    grid-template-rows: 180px auto 60px;
    font-family: sans-serif;
    text-align: center;
    background: url(${woodBackground});
    background-size: cover;
    height: 100vh;
    max-width: 600px;

    header {
        box-shadow: 0 5px 5px #ee000000;
    }

    h1 {
        color: white;
    }

    img {
        width: 45%;
        margin-top: 30px;
        margin-right: 30px;
    }

    section {
        overflow: auto;
    }

    footer {
        background: white;
    }
`
