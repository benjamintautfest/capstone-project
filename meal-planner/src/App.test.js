import App from './App'
import { render } from '@testing-library/react'
import user from '@testing-library/user-event'

describe('App', () => {
    it('renders correctly', () => {
        const { container } = render(<App />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('shows submenu item after button click', () => {
        const { getByText, queryByText, getByRole } = render(<App />)
        const button = getByRole('button', { name: /montag/i })
        expect(queryByText('Käsestulle')).not.toBeInTheDocument()

        user.click(button)
        const listitem = getByText('Käsestulle')

        expect(listitem).toBeInTheDocument()
    })
})
