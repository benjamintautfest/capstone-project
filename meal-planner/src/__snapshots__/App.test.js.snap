// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="App__AppStyled-c3tp1e-0 dduStM"
>
  <header>
    <img
      alt=""
      src="logo.svg"
    />
  </header>
  <section>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Montag"
      >
        <span
          class="day"
        >
          Montag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Dienstag"
      >
        <span
          class="day"
        >
          Dienstag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Mittwoch"
      >
        <span
          class="day"
        >
          Mittwoch
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Donnerstag"
      >
        <span
          class="day"
        >
          Donnerstag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Freitag"
      >
        <span
          class="day"
        >
          Freitag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Samstag"
      >
        <span
          class="day"
        >
          Samstag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
    <div>
      <button
        class="sc-bdfBwQ CGJgb"
        id="Sonntag"
      >
        <span
          class="day"
        >
          Sonntag
        </span>
        <span
          class="meal"
        >
          Auswählen
        </span>
      </button>
    </div>
  </section>
  <footer />
</div>
`;
